### PR TITLE
Path fix

### PR DESCRIPTION
**Pull request by worktycho**
A longer term fix to the problem with cPath assignment

### CODE REVIEW

**Comment by Seadragon91** [comment]
Tested it, works fine :smile:.

**Comment by Seadragon91** [comment]
@worktycho This is ready to be merged?

**Comment by SafwatHalaby** [comment]
Wait. Let me check it.

File: src/Mobs/Path.cpp [hunk 1]
```diff
@@ -497,3 +470,62 @@ cPathCell * cPath::GetCell(const Vector3i & a_Location)
...
+
+
+void cPath::Reset(
+	cChunk & a_Chunk,
+	const Vector3d & a_StartingPoint, const Vector3d & a_EndingPoint, int a_MaxSteps,
+	double a_BoundingBoxWidth, double a_BoundingBoxHeight,
+	int a_MaxUp, int a_MaxDown
+)
+{
+	m_StepsLeft = a_MaxSteps;
```
Comment (SafwatHalaby) [inline src/Mobs/Path.cpp]: Shouldn't we call `m_Map.clear()` and `m_OpenList.clear()` here?

File: src/Mobs/Path.h [hunk 2]
```diff
@@ -129,6 +136,13 @@ class cPath
 		return m_PathPoints.size() - m_CurrentPoint;
 	}
 
+	void Reset(
```
Comment (SafwatHalaby) [inline src/Mobs/Path.h]: Doxy docs. You can mostly copy the constructor docs.

File: src/Mobs/Path.h [hunk 3]
```diff
@@ -129,6 +137,31 @@ class cPath
 		return m_PathPoints.size() - m_CurrentPoint;
 	}
 
+	/** Creates a pathfinder instance. A Mob will probably need a single pathfinder instance for its entire life.
```
Comment (SafwatHalaby) [inline src/Mobs/Path.h]: Recreates?

**Comment by SafwatHalaby** [comment]
lgtm, except for the CircleCI errors.

**Comment by worktycho** [comment]
Updated.

File: src/Mobs/Path.h [hunk 4]
```diff
@@ -141,6 +174,10 @@ class cPath
 	void FinishCalculation(ePathFinderStatus a_NewStatus);  // Clears the memory used for calculating the path and changes the status.
 	void AttemptToFindAlternative();
 	void BuildPath();
+	void ResetImpl( //  Handles all logic associated with reseting the path to a clean state
```
Comment (madmaxoft) [inline src/Mobs/Path.h]: yuck

Comment (worktycho) [inline src/Mobs/Path.h]: My stupid misreading of the Stylechecker guidance. Fixed.

File: lib/cmake-coverage [hunk 5]
```diff
@@ -1 +1 @@
-Subproject commit fde8c1ef14684d763e7728dda98d49f2330534d1
+Subproject commit 25af3c657e42543e3e8b43378ebc5381e6260daa
```
Comment (SphinxC0re) [inline lib/cmake-coverage]: what did you do to the cmake-coverage?

Comment (worktycho) [inline lib/cmake-coverage]: I added a filter for the lib folder. We don't need coverage reports for the libs

**Comment by SafwatHalaby** [comment]
@worktycho Do you plan on introducing more changes here? If not, let us merge.

File: src/Mobs/Path.h [hunk 6]
```diff
@@ -141,6 +174,10 @@ class cPath
 	void FinishCalculation(ePathFinderStatus a_NewStatus);  // Clears the memory used for calculating the path and changes the status.
 	void AttemptToFindAlternative();
 	void BuildPath();
+	void ResetImpl(  // Handles all logic associated with reseting the path to a clean state
```
Comment (madmaxoft) [inline src/Mobs/Path.h]: This is still a yuck. Use a proper doxy-comment

File: lib/cmake-coverage (hunk 5 above)
Comment (worktycho) [inline lib/cmake-coverage]: This shouldn't be in this PR.

**Pull request merged**